skada/deep/base.py

```diff
@@ -497,7 +503,41 @@ def predict_features(self, X):
         if not torch.is_tensor(X):
             X = torch.tensor(X)
 
-        _, features = self.module_(
-            X, sample_domain=None, is_fit=False, return_features=True
-        )
-        return features
+
+        features_list = []
+        for features in self.feature_iter(X, training=False, return_features=True):
+            features = features[0] if isinstance(features, tuple) else features
```

Comment:
Why do you test if it is a tuple ? Right now It should return only the features, maybe I'm wrong. I'm working on a PR when now the features could also be a list of features. Let's see how it evolves